test/extended/cmd.sh

```diff
@@ -54,67 +54,68 @@ oadm policy add-role-to-group view system:authenticated -n default
+VERBOSE=true os::cmd::expect_success "oc new-project test-scratchimage"
+os::cmd::expect_success "oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker"
+os::cmd::expect_success_and_text "oc get bc ruby-hello-world -o jsonpath={.spec.strategy.dockerStrategy.from.kind}" "DockerImage"
+os::cmd::expect_success_and_text "oc get bc ruby-hello-world -o jsonpath={.spec.strategy.dockerStrategy.from.name}" "test/scratchimage:latest"
+os::cmd::expect_success "oc delete project test-scratchimage"
+VERBOSE=true os::cmd::expect_success "oc project new-app"
 # error due to partial match
-[ "$(oc new-app test/scratchimage2 -o yaml |& tr '\n' ' ' 2>&1 | grep -E "partial match")" ]
+os::cmd::expect_failure_and_text "oc new-app test/scratchimage2 -o yaml" "partial match"
 # success with exact match	
-[ "$(oc new-app test/scratchimage -o yaml)" ]
+os::cmd::expect_success "oc new-app test/scratchimage"
 echo "[INFO] newapp: ok"
 
 echo "[INFO] Running env variable expansion tests"
-oc new-project envtest
-oc create -f test/extended/fixtures/test-env-pod.json
-tryuntil "oc get pods | grep Running"
-podname=$(oc get pods --template='{{with index .items 0}}{{.metadata.name}}{{end}}')
-oc exec test-pod env | grep podname=test-pod
-oc exec test-pod env | grep podname_composed=test-pod_composed
-oc exec test-pod env | grep var1=value1
-oc exec test-pod env | grep var2=value1
-oc exec test-pod ps ax | grep "sleep 120"
+VERBOSE=true os::cmd::expect_success "oc new-project envtest"
+os::cmd::expect_success "oc create -f test/extended/fixtures/test-env-pod.json"
+os::cmd::try_until_text "oc get pods" "Running"
+os::cmd::expect_success_and_text "oc exec test-pod env" "podname=test-pod"
+os::cmd::expect_success_and_text "oc exec test-pod env" "podname_composed=test-pod_composed"
+os::cmd::expect_success_and_text "oc exec test-pod env" "var1=value1"
+os::cmd::expect_success_and_text "oc exec test-pod env" "var2=value1"
+os::cmd::expect_success_and_text "oc exec test-pod ps ax" "sleep 120"
 echo "[INFO] variable-expansion: ok"
 
 echo "[INFO] Running image pull secrets tests"
-oc login ${MASTER_ADDR} -u pull-secrets-user -p password --certificate-authority=${MASTER_CONFIG_DIR}/ca.crt
+VERBOSE=true os::cmd::expect_success "oc login '${MASTER_ADDR}' -u pull-secrets-user -p password --certificate-authority='${MASTER_CONFIG_DIR}/ca.crt'"
 
 # create a new project and push a busybox image in there
-oc new-project image-ns
+VERBOSE=true os::cmd::expect_success "oc new-project image-ns"
 os::cmd::expect_success "oc delete all --all"
-IMAGE_NS_TOKEN=$(oc sa get-token builder)
-os::cmd::expect_success "docker login -u imagensbuilder -p ${IMAGE_NS_TOKEN} -e fake@example.org ${DOCKER_REGISTRY}"
+os::cmd::expect_success "docker login -u imagensbuilder -p '$( oc sa get-token builder )' -e fake@example.org ${DOCKER_REGISTRY}"
```

Comment:
As mentioned, the quoting is something I'd like to do, but at the moment I'm trying to figure out which changes are causing the test failure, so I'm reverting things.